engines/sherlock/graphics.cpp

```diff
@@ -0,0 +1,205 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "sherlock/graphics.h"
+#include "sherlock/sherlock.h"
+#include "common/system.h"
+#include "graphics/palette.h"
+
+namespace Sherlock {
+
+Surface::Surface(uint16 width, uint16 height): _freePixels(true) {
+	create(width, height);
+}
+
+Surface::Surface(Surface &src, const Common::Rect &r) : _freePixels(false) {
+	setPixels(src.getBasePtr(r.left, r.top));
+	w = r.width();
+	h = r.height();
+	pitch = src.pitch;
+	format = Graphics::PixelFormat::createFormatCLUT8();
+}
+
+Surface::Surface() : _freePixels(false) {
+}
+
+Surface::~Surface() {
+	if (_freePixels)
+		free();
+}
+
+/**
+ * Sets up an internal surface with the specified dimensions that will be automatically freed
+ * when the surface object is destroyed
+ */
+void Surface::create(uint16 width, uint16 height) {
+	if (_freePixels)
+		free();
+
+	Graphics::Surface::create(width, height, Graphics::PixelFormat::createFormatCLUT8());
+	_freePixels = true;
+}
+
+/**
+ * Copy a surface into this one
+ */
+void Surface::blitFrom(const Graphics::Surface &src) {
+	blitFrom(src, Common::Point(0, 0));
+}
+
+/**
+ * Draws a surface at a given position within this surface
+ */
+void Surface::blitFrom(const Graphics::Surface &src, const Common::Point &pt) {
+	Common::Rect drawRect(0, 0, src.w, src.h);
+	Common::Point destPt = pt;
+
+	if (destPt.x < 0) {
+		drawRect.left += -destPt.x;
+		destPt.x = 0;
+	}
+	if (destPt.y < 0) {
+		drawRect.top += -destPt.y;
+		destPt.y = 0;
+	}
+	int right = destPt.x + src.w;
```

Comment:
Yes, it's correct. Say the dest point x = -5, then drawRect.left += -(-5), so it adds 5 onto the left. So essentially, it's drawing the sprite from the 5th column, since columns 0-4 are "off-screen". I'm not sure what you mean by the second comment about calling the next blitFrom().. I already am calling blitFrom if the drawRect that results is valid (i.e. there's something to display).